src/lib/containers/table/table-top/ColumnSelection.tsx

```diff
@@ -10,107 +10,88 @@ type ColumnSelectionProps = {
+    isColumnSelected,
+    visibleColumnCount = Infinity,
+    show,
+    onToggle,
+    toggleColumnSelection,
+  } = props
+  return (
+    <Dropdown
+      onToggle={(show: boolean, event: any, metadata: any) =>
+        onToggle(show, event, metadata)
+      }
+      show={show}
+    >
+      <Dropdown.Toggle
+        data-for={tooltipColumnSelectionId}
+        id={tooltipColumnSelectionId}
+        data-tip="Add / Remove Columns"
+        variant="light"
+      >
+        <img alt="columns selection" src={ColumnsSvg} />
+      </Dropdown.Toggle>
+      {/* There's a known issue if the number of dropdown items is very large, ~30+, the dropdown
           will unexpectedly render with the list going upwards instead of downwards.
         */}
-        <Dropdown.Menu
-          className="SRC-primary-color-hover-dropdown"
-          alignRight={true}
-        >
-          {headers.map((header, index) => {
-            let isCurrentColumnSelected: boolean | undefined =
-              isColumnSelected[index]
-            if (isCurrentColumnSelected === undefined) {
-              isCurrentColumnSelected = index < visibleColumnCount
-            }
-            const iconStyle: any = { width: '11px', marginRight: '10px' }
-            if (!isCurrentColumnSelected) {
-              iconStyle.visibility = 'hidden'
-            }
-            const maybeShowPrimaryColor = isCurrentColumnSelected
-              ? 'SRC-primary-text-color'
-              : ''
-            return (
-              <Dropdown.Item
-                onClick={() => toggleColumnSelection(index)()}
-                key={header.name}
-              >
-                <FontAwesomeIcon
-                  style={iconStyle}
-                  className={maybeShowPrimaryColor}
-                  icon="check"
-                />
-                {header.name}
-              </Dropdown.Item>
-            )
-          })}
-        </Dropdown.Menu>
-        <ReactTooltip
-          delayShow={TOOLTIP_DELAY_SHOW}
-          place="top"
-          type="dark"
-          effect="solid"
-          id={tooltipColumnSelectionId}
-        />
-      </Dropdown>
-    )
-  }
+      <Dropdown.Menu
+        className="SRC-primary-color-hover-dropdown"
+        alignRight={true}
+      >
+        {headers.map((header, index) => {
+          let isCurrentColumnSelected: boolean | undefined =
+            isColumnSelected[index]
+          if (isCurrentColumnSelected === undefined) {
+            isCurrentColumnSelected = index < visibleColumnCount
+          }
+          const iconStyle: any = { width: '11px', marginRight: '10px' }
+          if (!isCurrentColumnSelected) {
+            iconStyle.visibility = 'hidden'
+          }
+          const maybeShowPrimaryColor = isCurrentColumnSelected
+            ? 'SRC-primary-text-color'
+            : ''
+          return (
+            // @ts-ignore
+            <Dropdown.Item
+              onClick={(event: any) => toggleColumnSelection(index)(event)}
```

Comment:
a little unclear of whey are we curring here

Reply:
good call, its not needed, will remove